daffodil-test/src/test/resources/org/apache/daffodil/extensions/type_calc/inputTypeCalc.tdml

```diff
@@ -50,6 +50,8 @@
       </xs:complexType>
     </xs:element>
 
+    <xs:element name="keysetValue_03" type="tns:keysetValue_01" dfdl:lengthKind="implicit"/>
```

Comment:
So is this the workaround? Wherever you use a type where the type carries a dfdlx:repType you need this dfdl:lengthKind 'implicit' ?

How come the other tests in here didn't need this?

Please add doc/comments explaining this and include the DAFFODIL-NNNN ticket number about the issue.

Reply:
Will do.  Not a workaround, merely setting it to implicit to trigger the bug that the code changes fixed.  The problem had been that it was thinking that the keysetValue_01 type was a string, which when lengthKind="implicit" causing it to complain about a missing maxLength property as you can't have an implicit length string without one.

This was one of the things we were missing when we were trying to debug the P8 schema yesterday.  The tests that were working had a lengthKind="delimited", which was fine given that it thought it was processing a string.  Adding dfdl:lengthKind="implicit" seemed like the easiest way to trigger the bug we were seeing in the P8 schema.

I'll work on adding some comments and a better name for the test now.